xds/internal/client/client.go

```diff
@@ -328,16 +388,27 @@ type clientImpl struct {
 
 	logger *grpclog.PrefixLogger
 
-	updateCh    *buffer.Unbounded // chan *watcherInfoWithUpdate
+	updateCh *buffer.Unbounded // chan *watcherInfoWithUpdate
+	// All the following maps are to keep the updates/metadata in a cache.
+	// TODO: move them to a separate struct/package, to cleanup the xds_client.
+	// And CSDS handler can be implemented directly by the cache.
 	mu          sync.Mutex
 	ldsWatchers map[string]map[*watchInfo]bool
+	ldsVersion  string
```

Comment:
Is this version field only used in the CSDS response? Can we add a small comment for the same. Thanks.

Reply:
Yes, comment added.

Actually, only LDS needs the version :man_shrugging:
But I added version for the other resources as well.